tools/gen-device-svd/gen-device-svd.go

```diff
@@ -342,6 +342,11 @@ func readSVD(path, sourceURL string) (*Device, error) {
 				firstAddress := clusterRegisters[0].address
 				dimIncrement = int(lastAddress - firstAddress)
 			}
+
+			if !unicode.IsUpper(rune(clusterName[0])) && !unicode.IsDigit(rune(clusterName[0])) {
```

Comment:
Strictly speaking the cluster name cannot start with a digit, or it would be an invalid identifier in most major programming languages.